### PR TITLE
add carousel in Facebook channel

### PR DESCRIPTION
**Pull request by veptechno**
Implement carousel functionality in FacebookChannel and add a general CarouselReaction to the core that is used in FacebookChannel, ChatWidgetChannel and AliceChannel.
JAICF-67

### CODE REVIEW

File: channels/yandex-alice/src/main/kotlin/com/justai/jaicf/channel/yandexalice/api/AliceApi.kt [hunk 1]
```diff
@@ -53,7 +53,9 @@ class AliceApi(
 
     fun getImageId(url: String) = images.getOrPut(url) { uploadImage(url).id }
 
-    fun uploadImage(url: String): Image = runBlocking {
+    fun getImageUrl(id: String) = images.entries.find { id == it.value }?.key
```
Comment (Denire) [inline channels/yandex-alice/src/main/kotlin/com/justai/jaicf/channel/yandexalice/api/AliceApi.kt:56]: I suggest making this api method internal. It's used only for our internal logic.

Comment (veptechno) [reply channels/yandex-alice/src/main/kotlin/com/justai/jaicf/channel/yandexalice/api/AliceApi.kt:56]: Good suggestion

File: channels/yandex-alice/src/main/kotlin/com/justai/jaicf/channel/yandexalice/api/AliceApi.kt [hunk 2]
```diff
@@ -53,7 +53,9 @@ class AliceApi(
...
+
+    private fun uploadImage(url: String): Image = runBlocking {
```
[the remark this answers is not in the document]
Comment (veptechno) [reply channels/yandex-alice/src/main/kotlin/com/justai/jaicf/channel/yandexalice/api/AliceApi.kt:58]: Ok

File: channels/yandex-alice/src/main/kotlin/com/justai/jaicf/channel/yandexalice/api/AliceApi.kt [hunk 3]
```diff
@@ -62,7 +64,7 @@ class AliceApi(
         imageStorage[skillId]?.put(image.origUrl, image.id)
     }
 
-    fun listImages(): List<Image> = runBlocking {
+    private fun listImages(): List<Image> = runBlocking {
```
[the remark this answers is not in the document]
Comment (veptechno) [reply channels/yandex-alice/src/main/kotlin/com/justai/jaicf/channel/yandexalice/api/AliceApi.kt:67]: Ok

**Review by Denire — APPROVED** [review]

**Pull request merged**